### PR TITLE
Update integration tests

### PR DESCRIPTION
**Pull request by vedantchokshi**
Fixes #

### CODE REVIEW

File: beekeeper-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/apiary/generator/HousekeepingEntityGeneratorTestBase.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ protected void setupClockAndExtractor(ListenerEvent listenerEvent) {
 
   protected void assertHousekeepingEntity(HousekeepingEntity housekeepingEntity,
       LifecycleEventType lifecycleEventType) {
+    System.out.println(housekeepingEntity);
```
[the remark this answers is not in the document]
Comment (vedantchokshi) [reply beekeeper-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/apiary/generator/HousekeepingEntityGeneratorTestBase.java:65]: :o

Comment (vedantchokshi) [inline beekeeper-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/apiary/generator/HousekeepingEntityGeneratorTestBase.java:65]:
```suggestion
    System.out.println(housekeepingEntity);
```

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperCleanupIntegrationTest.java [hunk 2]
```diff
@@ -49,107 +53,77 @@
...
+  public static void teardown() {
     amazonS3.shutdown();
-    mySqlTestUtils.close();
-    s3Container.stop();
-    mySQLContainer.stop();
   }
 
   @BeforeEach
-  void setup() throws SQLException {
+  public void setup() {
```
Comment (patduin) [inline beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperCleanupIntegrationTest.java:102]: these non static methods need to be public as well?

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,13 +8,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Add Time To Live (TTL) for all tables.
 - DB migration for creating new `housekeeping_metadata` table.
+- Integration test for tables with TTL.
```
Comment (massdosage) [inline CHANGELOG.md:11]: Since most end users don't really care about or use the tests we generally don't mention changes to them in the CHANGELOG.

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java [hunk 4]
```diff
@@ -37,8 +37,9 @@
   @Query(value = "from HousekeepingMetadata t "
       + "where t.databaseName = :databaseName "
       + "and t.tableName = :tableName "
-      + "and t.partitionName = :partitionName "
+      + "and (t.partitionName = :partitionName or t.partitionName is NULL) " // To handle special null case
```
Comment (massdosage) [inline beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java:40]: This is a user facing change so could be worth mentioning unless we think it's such an edge case that it's not.

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperExpiredMetadataSchedulerApiaryIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+  private static final String LOCATION_B = "s3://location-b";
+
+  @Container
+  private static final LocalStackContainer SQS_CONTAINER = ContainerTestUtils.awsContainer(SQS);
+  private static AmazonSQS amazonSQS;
+
+  @BeforeAll
+  public static void init() {
+    String queueUrl = ContainerTestUtils.queueUrl(SQS_CONTAINER, QUEUE);
+    System.setProperty("properties.apiary.queue-url", queueUrl);
```
Comment (massdosage) [inline beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperExpiredMetadataSchedulerApiaryIntegrationTest.java:92]: Should we not unset this in the `@AfterAll` method so as not to "infect" other tests?

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperIntegrationTestBase.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    String jdbcUrl = MY_SQL_CONTAINER.getJdbcUrl() + "?useSSL=false";
+    String username = MY_SQL_CONTAINER.getUsername();
+    String password = MY_SQL_CONTAINER.getPassword();
+
+    System.setProperty("spring.datasource.url", jdbcUrl);
+    System.setProperty("spring.datasource.username", username);
+    System.setProperty("spring.datasource.password", password);
+    System.setProperty("aws.region", AWS_REGION);
+    System.setProperty("aws.accessKeyId", AWS_ACCESS_KEY_ID);
+    System.setProperty("aws.secretKey", AWS_SECRET_KEY);
```
Comment (massdosage) [inline beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperIntegrationTestBase.java:113]: Same as my other comment, we often use https://stefanbirkner.github.io/system-rules/ to manage anything to do with the `System` calls so it might be useful here.

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/CommonTestVariables.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.integration;
+
+import static java.time.ZoneOffset.UTC;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+public final class CommonTestVariables {
```
Comment (massdosage) [inline beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/CommonTestVariables.java:23]: You should probably give this a private constructor to prevent anyone from trying to instantiate it. It's not so important for a test but it's good practice to think about if you need to do something like this in production code.

File: beekeeper-integration-tests/src/test/resources/message.json [hunk 8]
```diff
@@ -7,5 +7,5 @@
   "Signature": "signature",
   "SigningCertURL": "https://sns.us-west-2.amazonaws.com/SimpleNotificationService-xxxx",
   "UnsubscribeURL": "https://sns.us-west-2.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-west-2:440407435941:sns-topic",
-  "Message": "{ \"protocolVersion\": \"1.0\", \"eventType\": \"DROP_TABLE\", \"dbName\": \"some_db\", \"tableName\": \"some_table\", \"tableLocation\": \"%s\", \"tableParameters\": { \"beekeeper.unreferenced.data.retention.period\": \"P7D\", \"beekeeper.remove.unreferenced.data\": \"%s\",  \"beekeeper.hive.event.whitelist\":\"%s\" }}"
+  "Message": "replace"
```
Comment (massdosage) [inline beekeeper-integration-tests/src/test/resources/message.json:10]: Up to you, but it be clearer for the string to replace to be something like `${replace}` as that's a convention used in quite a few templating libraries which is the intention here.

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java (hunk 4 above)
Comment (vedantchokshi) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java:40]: It's part of the new stuff we have added so users aren't using this. It's just something I missed which was caught in the integration tests. Do you think I should add it to the CHANGELOG?

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperExpiredMetadataSchedulerApiaryIntegrationTest.java (hunk 5 above)
Comment (vedantchokshi) [reply beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperExpiredMetadataSchedulerApiaryIntegrationTest.java:92]: Ah I never thought of that. I will do that

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java (hunk 4 above)
Comment (massdosage) [reply beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java:40]: No, if it's new and part of the broader changes which are already described in the CHANGELOG then it's fine.

File: CHANGELOG.md (hunk 3 above)
Comment (massdosage) [reply CHANGELOG.md:11]: Ping ^

File: beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/ExpiredHousekeepingMetadataGenerator.java [hunk 9]
```diff
@@ -45,8 +48,10 @@
 
 public class ExpiredHousekeepingMetadataGenerator implements HousekeepingEntityGenerator {
 
+  private static final Logger log = LoggerFactory.getLogger(UnreferencedHousekeepingPathGenerator.class);
```
Comment (JayGreeeen) [inline beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/ExpiredHousekeepingMetadataGenerator.java:51]: This should be `ExpiredHousekeepingMetadataGenerator.class`

Comment (massdosage) [reply beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/ExpiredHousekeepingMetadataGenerator.java:51]: Niiiice! You've just earned your "hawk eyes" badge @JayGreeeen!

**Review by massdosage — APPROVED** [review]

**Review by JayGreeeen — APPROVED** [review]
